ykrt/src/mt.rs

```diff
@@ -255,69 +226,69 @@ impl MTThread {
         // to update values, allowing us to detect if we were interrupted. If we were interrupted,
         // we simply retry the whole operation.
         loop {
-            let pack = &loc.inner.pack;
             // We need Acquire ordering, as PHASE_COMPILED will need to read information written to
             // external data as a result of the PHASE_TRACING -> PHASE_COMPILED transition.
-            let lp = pack.load(Ordering::Acquire);
+            let lp = loc.pack.load(Ordering::Acquire);
             match lp & PHASE_TAG {
                 PHASE_COUNTING => {
                     let count = (lp & !PHASE_TAG) >> 2;
-                    let new_pack;
                     if count >= self.inner.hot_threshold {
                         if self.inner.tracer.is_some() {
                             // This thread is already tracing. Note that we don't increment the hot
                             // count further.
-                            break;
+                            return None;
                         }
-                        if pack.compare_and_swap(lp, PHASE_TRACING, Ordering::Release) == lp {
+                        let loc_id = Box::into_raw(Box::new(0u8));
```

Comment:
Should generation and fetching of IDs be encapsulated inside `Location` to allow the implementation to change?

Reply:
We should also document the purpose of the box allocation, as it's non-obvious.

Reply:
> We should also document the purpose of the box allocation, as it's non-obvious.

I see, it's documented in `MTThreadInner`. OK!